packages/binding-coap/src/coap-server.ts

```diff
@@ -194,25 +194,21 @@ export default class CoapServer implements ProtocolServer {
         });
     }
 
-    public destroy(thingId: string): Promise<boolean> {
+    public async destroy(thingId: string): Promise<boolean> {
         debug(`CoapServer on port ${this.getPort()} destroying thingId '${thingId}'`);
-        return new Promise<boolean>((resolve, reject) => {
-            let removedThing: ExposedThing;
-            for (const name of Array.from(this.things.keys())) {
-                const expThing = this.things.get(name);
-                if (expThing?.id === thingId) {
-                    this.things.delete(name);
-                    this.coreResources.delete(name);
-                    removedThing = expThing;
-                }
-            }
-            if (removedThing) {
-                info(`CoapServer succesfully destroyed '${removedThing.title}'`);
-            } else {
-                info(`CoapServer failed to destroy thing with thingId '${thingId}'`);
+        for (const name of Array.from(this.things.keys())) {
```

Comment:
Having another look at the code, I think this could further be simplified like so:

```suggestion
        for (const name of this.things.keys()) {
```

(I can't test it out locally at the moment, therefore I left this comment as a note for later.)

Reply:
Tried on node console, `Array.from` seems completely redundant.

Reply:
Thank you for confirming! :) Then I will apply this suggestion and you could squash and merge this PR :)

Reply:
The commits are now already squashed, I made a minor change to one variable name, the PR should now be ready :)